### DISCUSSION TITLE
TextInput - rightSection -- fixed width?

### DISCUSSION THREAD

**Discussion by cstrat** [mantinedev/mantine#36] [Q&A] [answered]
Just asking the question because there might be a good answer - or I may be doing this incorrectly.

Why does the `TextInput` `rightSection` node require a fixed width?
Wouldn't using Flexbox on the wrapper with the input element having flex-grow allow for any sized `rightSection`?

I am wanting to embed a `<Button>` to save the field on a form (individual fields rather than saving the whole form).
However if I want the `<Button>` text to be dynamic (or even just behave with different fonts/sizing) having a fixed `rightSectionWidth` is troublesome.

Unsure but maybe there is something I can do by passing `rightSectionProps`. Haven't really had a chance to check that out yet.

https://codesandbox.io/s/mantine-input-rightselection-yx1n4?file=/src/App.js

**Comment by rtivital** [comment, marked as answer]
Flexbox won't work here, right section has [position absolute](https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/Input/Input.styles.ts#L179-L187), for example, in docs search, you can click the keyboard shortcut and it will still move focus the search field (with `rightSectionProps={{ style: { pointerEvents: 'none' } }}`). Dynamic width can only be achieved with getBoundingClientRect, but it won't work during ssr which will result in jumping input layout. So I guess, `rightSectionWidth` is necessary here.

**Reply by cstrat** [reply]
Thanks for the explanation. I will just stick with an `<ActionIcon>` for now then.

**Reply by ifeltsweet** [reply]
Just for reference, MUI uses flex positioning for the input and its adornments (right/left sections) so this works out of the box.
